src/test/java/com/baeldung/selenium/alertsandpopups/HandlePopupsUnitTests.java

```diff
@@ -0,0 +1,57 @@
+package com.baeldung.selenium.alertsandpopups;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.time.Duration;
+
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class HandlePopupsUnitTests {
+
+    @Test
+    public void whenBasicAuthPopupAppears_thenBypassWithCredentials() {
+        WebDriver driver = new ChromeDriver();
+
+        String username = "admin";
+        String password = "admin";
+        String url = "https://" + username + ":" + password + "@the-internet.herokuapp.com/basic_auth";
+
+        driver.get(url);
+
+        String bodyText = driver.findElement(By.tagName("body"))
+            .getText();
+        assertTrue(bodyText.contains("Congratulations! You must have the proper credentials."));
+
+        driver.quit();
```

Comment:
This should be common to all tests (in an after block) to ensure the driver is quit even if the test fails